src/Compilers/CSharp/Portable/BoundTree/Statement.cs

```diff
@@ -9,23 +9,51 @@
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
-    internal partial class BoundStatement : IOperation
+    internal partial class BoundNode : IOperation, IOperationWithChildren
     {
-        OperationKind IOperation.Kind => this.StatementKind;
+        OperationKind IOperation.Kind => this.OperationKind;
 
         bool IOperation.IsInvalid => this.HasErrors;
 
         SyntaxNode IOperation.Syntax => this.Syntax;
 
-        ITypeSymbol IOperation.Type => null;
+        ITypeSymbol IOperation.Type => OperationType;
 
-        Optional<object> IOperation.ConstantValue => default(Optional<object>);
+        Optional<object> IOperation.ConstantValue => this.OperationConstantValue;
+
+        ImmutableArray<IOperation> IOperationWithChildren.Children => this.Children;
+
+        protected virtual OperationKind OperationKind => OperationKind.None;
+
+        protected virtual ITypeSymbol OperationType => null;
+
+        protected virtual Optional<object> OperationConstantValue => default(Optional<object>);
+
+        /// <summary>
+        /// Override this property to return the child operations if the IOperation API corresponding to this bound node is not yet designed or implemented.
+        /// </summary>
+        protected virtual ImmutableArray<IOperation> Children => ImmutableArray<IOperation>.Empty;
+
+        public virtual void Accept(OperationVisitor visitor)
+        {
+            visitor.VisitNoneOperation(this);
+        }
+
+        public virtual TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
```

Comment:
Never mind, looks fine.

---
In reply to: [113977715](https://github.com/dotnet/roslyn/pull/18976#discussion_r113977715) [](ancestors = 113977715)